src/main/java/gift/product/service/ProductService.java

```diff
@@ -70,8 +67,13 @@ public Product findById(Long id){
      * @param requestDto
      */
     public void update(Long id, ProductRequestDto requestDto){
-        Product product = findById(id);
-        product.update(requestDto.name(), requestDto.price(), requestDto.imageUrl());
+        Optional<Product> product = repository.findById(id);
+
+        if (product.isEmpty()){
+            throw new ProductNotFoundException("상품을 찾을 수 없습니다. ID: " + id);
+        }
```

Comment:
```suggestion
        return repository.findById(id)
                .orElseThrow(() -> new ProductNotFoundException("상품을 찾을 수 없습니다. ID: " + id));
```

이런식으로도 사용할 수 있습니다! Optional은 다양한 메서드를 지원해주니 알아두면 좋습니다👍 참고 링크 드립니다:)
https://tecoble.techcourse.co.kr/post/2021-06-20-optional-vs-null/